metricq/history_client.py

```diff
@@ -327,8 +331,8 @@ async def _history_consume(self, extra_queues=[]):
             loop=self.event_loop,
         )
 
-    async def _on_history_response(self, message: aio_pika.Message):
-        with message.process(requeue=True):
+    async def _on_history_response(self, message: aio_pika.IncomingMessage):
+        async with message.process(requeue=True):
```

Comment:
how did this ever work o.O?

Reply:
In an old version of aio-pika this was a synchronous context manager

Reply:
Can you pinpoint that? We do pin a specific minor version of pika, so did they change that in a patch? We should at least make a dependency such that it is not satisfied by a version that still has the synchronous context manager.

Reply:
Okay, it's not dropped but deprecated https://github.com/mosquito/aio-pika/blob/master/aio_pika/message.py#L714

Reply:
I see, and the `async` one was added ages ago already: https://github.com/mosquito/aio-pika/blob/master/CHANGELOG.md#290 . So we should be just fine with the change :)